### PR TITLE
Codecov Report Publishing

### PR DESCRIPTION
**Pull request by gayaldassanayake**
## Purpose
- Introducing `Codecov` code coverage report publishing to the graphql repository

## Goals
- Generating a code coverage report that is published via `Codecov` triggered via GitHub actions of build and PR.
- `Codecov` bot commenting on PRs

## Approach
-  Custom execution file, java and ballerina class files, Sourcecode is provided to `jacoco`  to generate an XML code coverage report.
- A new job is inserted to `Build` an `Pull request` GitHub actions that publishes the XML report to  `Codecov`.
- These jobs trigger on merge to master or pull request and CodeCov generates a descriptive code coverage report.
- `Codecov` also adds comments to any PR made based on the code coverage changes between commits

### CODE REVIEW

File: .github/workflows/build-master.yml [hunk 1]
```diff
@@ -39,6 +40,11 @@ jobs:
                     name: Code Coverage JSON
                     path: graphql-ballerina/target/report/test_results.json
                     if-no-files-found: ignore
+            -   name: Generate CodeCov Report
+                uses: codecov/codecov-action@v1
+                with:
+                    token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (azinneera) [inline .github/workflows/build-master.yml:46]: Do we need the code cov token?

File: build.gradle [hunk 2]
```diff
@@ -229,3 +230,26 @@ subprojects {
...
+        xml.enabled = true
+        html.enabled = true
+        xml.destination = new File("${buildDir}/reports/jacoco/report.xml")
+        html.destination = new File("${buildDir}/reports/jacoco/report.html")
+    }
+
+    onlyIf = {
+        true
+    }
+}
```
Comment (azinneera) [inline build.gradle:255]: Missing new line

File: codecov.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+fixes:
+  - "ballerina/graphql/*/::graphql-ballerina/"
+  - "ballerina/graphql$0046parser/*::graphql-ballerina/modules/parser"
+
+ignore:
+  - "**/test"
```
Comment (azinneera) [inline codecov.yml:6]: Missing new line

File: .github/workflows/build-master.yml (hunk 1 above)
Comment (gayaldassanayake) [reply .github/workflows/build-master.yml:46]: Found in Documentation that its not need for public repos. Will test and change

File: graphql-native/build.gradle [hunk 4]
```diff
@@ -70,3 +70,11 @@ compileJava {
...
     }
 }
+
+task copyJavaClassFiles(type: Copy) {
+    from("${project.buildDir}/classes") {
+        exclude '**/module-info.class'
+        include '**/*.class'
+    }
+    into "${project.rootDir.absolutePath}/build/classes"
+}
```
Comment (azinneera) [inline graphql-native/build.gradle:80]: Missing new line

**Comment by azinneera** [comment]
LGTM

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**